libraries/engage/product/components/Swatches/Swatches.jsx

```diff
@@ -30,47 +23,41 @@ const SwatchesUnwrapped = ({ productId, characteristics, widgetPath }) => {
     return null;
   }
 
-  const settings = useWidgetSettings(widgetId, widgetPath);
+  const settings = useWidgetSettings(WIDGET_ID);
+
+  let swatches = characteristics.filter(c => c.swatch === true);
+  if (settings.filter && settings.filter.length) {
+    swatches = swatches.filter(swatch => settings.filter.includes(swatch.id));
+  }
+  if (!swatches) {
+    return null;
+  }
 
-  // override default settings with configured widget settings
-  const widgetSettings = defaultsDeep(settings, widgetDefaultSettings);
+  if (settings.maxItemCount) {
+    swatches = swatches.map(swatch => ({
+      ...swatch,
+      values: swatch.values.slice(0, settings.maxItemCount),
+    }));
+  }
 
-  const testId = `productSwatches.${productId}`;
   return (
     <SurroundPortals portalName={PRODUCT_SWATCHES} portalProps={{ characteristics }}>
-      { widgetSettings.maxItemCount > 0 && (
-        <div data-test-id={testId} className={swatches}>
-          {characteristics
-            .filter(c => c.swatch === true)
-            .filter(c => !widgetSettings.filter.length || widgetSettings.filter.includes(c.id))
-            .map(swatch => (
-              <Swatch
-                key={`${productId}.${swatch.id}`}
-                testId={`${testId}.swatch.${swatch.id}`}
-                swatch={swatch}
-                maxItemCount={widgetSettings.maxItemCount}
-              />
+      <div className={swatchesClass}>
+        {swatches.map(swatch => (
+          <Swatch key={`${productId}.${swatch.id}`} swatch={swatch} />
           ))}
-        </div>
-      )}
+      </div>
     </SurroundPortals>
   );
 };
 
-SwatchesUnwrapped.propTypes = {
+Swatches.propTypes = {
   productId: PropTypes.string.isRequired,
   characteristics: PropTypes.arrayOf(PropTypes.shape()),
-  widgetPath: PropTypes.oneOfType([
-    PropTypes.string,
-    PropTypes.number,
-  ]),
 };
 
-SwatchesUnwrapped.defaultProps = {
+Swatches.defaultProps = {
   characteristics: null,
-  widgetPath: undefined,
 };
 
-SwatchesUnwrapped.displayName = 'Swatches';
-
-export const Swatches = connect(SwatchesUnwrapped);
+export default memo(connect(Swatches));
```

Comment:
You should turn that around: `connect(memo())` because the memo function considers the incoming props. in this way, the connector add props AFTER the memoization happened. That could lead to unwanted side effects and issues.